corehq/apps/linked_domain/remote_accessors.py

```diff
@@ -121,13 +121,18 @@ def fetch_remote_media(local_domain, missing_media, remote_app_details):
         media_class = CommCareMultimedia.get_doc_class(item['media_type'])
         content = _fetch_remote_media_content(item, remote_app_details)
         media_item = media_class.get_by_data(content)
-        media_item._id = item['multimedia_id']
+        if getattr(media_item, '_id', None) and not getattr(item, 'upstream_media_id', None):
```

Comment:
`and not getattr(item, 'upstream_media_id', None)`

Just noting that this condition shouldn't be necessary since if the `item` did have an `upstream_media_id` it indicates that the media exists locally already and it shouldn't be in the list of `missing_media`.

Fine to keep this condition as a defensive guard but we should also be testing that the `_get_missing_multimedia` is working correctly and not returning media that exists locally.